types/global.d.ts

```diff
@@ -0,0 +1,4 @@
+declare type Falsy = false | null | undefined | '' | 0;
+declare type Maybe<X> = X extends Falsy ? never : X | Falsy;
+declare type Some<X> = X extends Falsy ? never : X;
+declare type DefinitelyHas<X extends {}, Keys> = X & { [k in keyof X & Keys]: Some<X[k]> };
```

Comment:
I thought for 5-10 minutes and couldn't come up with anything better. `Includes` is more concise but TypeScript already has an `Exclude` type which is meant to work with unions, so that might be confusing.